### PR TITLE
UI tests for app password

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
UI tests for app password.

## Motivation and Context
testing if adding app password and deleting app password works

## How Has This Been Tested?
Run test locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php [hunk 1]
```diff
@@ -101,4 +112,65 @@ public function theUserDisplayNameAndAppPasswordShouldBeDisplayedOnTheWebUI() {
...
 
+	/**
+	 * @When the user re-logs in with username :username and generated app password using the webUI
+	 * 
+	 * @param string $username
+	 * 
+	 * @return void
+	 */
+	public function theUserLogsInWithNewAppPassword($username) {
+		$password = $this->newAppPassword;
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php]: This line is not needed.

File: tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php [hunk 2]
```diff
@@ -34,17 +36,23 @@ class WebUIPersonalSecuritySettingsContext extends RawMinkContext implements Con
 	private $personalSecuritySettingsPage;
 	private $appName;
 	private $strForAppName = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+	private $featureContext;
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php]: ``featureContext`` seems to not be used?

File: tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature [hunk 3]
```diff
@@ -4,9 +4,29 @@ As a user
 I want to be able to manage security settings for my account
 So that I can enable, allow and deny access to and from other storage systems or resources
 
+	Background:
+			Given these users have been created but not initialized:
+			|username|password|displayname|email       |
+			|user1   |1234    |User One   |u1@oc.com.np|
+			And the user has browsed to the login page
+			When the user logs in with username "user1" and password "1234" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature]: This is part of the ``Given`` section. So use the form:
``And the user has logged in with``...

File: tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature [hunk 4]
```diff
@@ -4,9 +4,29 @@ As a user
...
+		And the user browses to the personal security settings page
 		When the user creates a new App password using the webUI
-		Then the new app should be listed in the App passwords list on the webUI
-		And the user display name and the app password should be displayed on the webUI
\ No newline at end of file
+		Then the user display name and the app password should be displayed on the webUI
+		And the new app should be listed in the App passwords list on the webUI
+
+	Scenario: login with new app password
+		And the user browses to the personal security settings page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature]: ``When the user``...

File: tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature [hunk 5]
```diff
@@ -4,9 +4,29 @@ As a user
...
+		And the user creates a new App password using the webUI
+		And the user re-logs in with username "user1" and generated app password using the webUI
+		Then the user should be redirected to a webUI page with the title "Files - ownCloud"
+
+	Scenario: delete the app password 
+		And the user browses to the personal security settings page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature]: ``When the user``...

File: tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature [hunk 6]
```diff
@@ -4,9 +4,29 @@ As a user
...
+			|username|password|displayname|email       |
+			|user1   |1234    |User One   |u1@oc.com.np|
+			And the user has browsed to the login page
+			When the user logs in with username "user1" and password "1234" using the webUI
+			
 	Scenario: create new app password
-		Given user admin has logged in using the webUI
-		And the user has browsed to the personal security settings page
+		Then the user should be redirected to a webUI page with the title "Files - ownCloud"
+		And the user browses to the personal security settings page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature]: ``When`` can go here.
and the next line is ``And``

File: tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature [hunk 7]
```diff
@@ -4,9 +4,29 @@ As a user
...
+			Given these users have been created but not initialized:
+			|username|password|displayname|email       |
+			|user1   |1234    |User One   |u1@oc.com.np|
+			And the user has browsed to the login page
+			When the user logs in with username "user1" and password "1234" using the webUI
+			
 	Scenario: create new app password
-		Given user admin has logged in using the webUI
-		And the user has browsed to the personal security settings page
+		Then the user should be redirected to a webUI page with the title "Files - ownCloud"
```
Comment (phil-davis) [inline tests/acceptance/features/webUIPersonalSettings/personalSecuritySettings.feature]: Is there a reason that this extra check is here?

**Review by phil-davis — COMMENTED** [review]
Mostly comment relating to keeping the ``Given`` and ``When`` in the usual places. Actually for test execution it makes no difference. But for human reading it is easier.

**Comment by paurakhsharma** [comment]
Yes, thank you for the review and suggestions. I will fix it right away.

File: tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php (hunk 1 above)
Comment (paurakhsharma) [reply tests/acceptance/features/bootstrap/WebUIPersonalSecuritySettingsContext.php]: Yes forgot to remove it.

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport is in #31218

**Pull request merged**